s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -95,11 +95,14 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
     val s3Headers = S3Headers(sse.fold[Seq[HttpHeader]](Seq.empty) { _.headersFor(GetObject) })
     val future = request(s3Location, rangeOption = range, versionId = versionId, s3Headers = s3Headers)
       .map(response => response.withEntity(response.entity.withoutSizeLimit))
+      .flatMap(entityForSuccess)
     val source = Source
-      .fromFuture(future.flatMap(entityForSuccess).map(_._1))
```

Comment:
I don't see what's changing here. Aren't you just changing `val future` to include `flatMap(entityForSuccess)`, rather than doing it in-line in the source declaration?

Reply:
@jypma , this affects the error handling for the object metadata future as well, it will now fail on failed responses (i.e.404)

Reply:
Ahh I see. So the effect of the change is not here, but in the line 103 which maps `future` which now has error handling because of the added `.flatMap(entityForSuccess)`.

Reply:
exactly